pkg/apis/networking/v1/types.go

```diff
@@ -174,3 +175,144 @@ type NetworkList struct {
 
 	Items []Network `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// RemoteCluster is a specification for a RemoteCluster resource
+type RemoteCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RemoteClusterSpec   `json:"spec"`
+	Status RemoteClusterStatus `json:"status"`
+}
+
+// RemoteClusterSpec is the spec for a RemoteCluster resource
+type RemoteClusterSpec struct {
+	ConnConfig APIServerConnConfig `json:"connConfig"`
+}
+
+type APIServerConnConfig struct {
+	// apiServer address. Format: https://ip:port
+	Endpoint   string `json:"endpoint"`
+	CABundle   []byte `json:"caBundle"`
+	ClientCert []byte `json:"clientCert"`
+	ClientKey  []byte `json:"clientKey"`
+	// The maximum length of time to wait before giving up on a server
+	// request. A value of zero means no timeout. Default: zero second
+	Timeout uint32 `json:"timeout"`
+}
+
+// RemoteClusterStatus is the status for a RemoteCluster resource
+type RemoteClusterStatus struct {
+	// Conditions is an array of current cluster conditions.
+	Conditions []ClusterCondition `json:"conditions"`
+	// If network in remote cluster is all underlay, it is nil
+	TunnelNetID *uint32 `json:"tunnelNetID,omitempty"`
```

Comment:
why this field in status? how to fetch it?

Reply:
List from Remote cluster's overlay network. Every Health check will do the job.

Reply:
change this to slice